base/float.jl

```diff
@@ -386,10 +386,35 @@ end
 end
 
 # fused multiply-add
-fma_libm(x::Float32, y::Float32, z::Float32) =
-    ccall(("fmaf", libm_name), Float32, (Float32,Float32,Float32), x, y, z)
-fma_libm(x::Float64, y::Float64, z::Float64) =
-    ccall(("fma", libm_name), Float64, (Float64,Float64,Float64), x, y, z)
+function fma_libm(x::Float32, y::Float32, z::Float32)
+    r = ccall(("fmaf", libm_name), Float32, (Float32,Float32,Float32), x, y, z)
+    # This is necessary on 32-bit Intel Linux, probably on OS X, and
+    # maybe on Windows
+    # TODO: Disable this for non-Intel systems
+    if is(Int, Int32)
```

Comment:
Are you sure that `Base.ARCH` is this complicated? For me it's just `:x86_64` -- the actual CPU is not encoded.

Are you using a 32-bit Linux system? What is the value of `Base.ARCH` for you?